refract-spec.md

```diff
@@ -12,89 +12,286 @@ This document conforms to RFC 2119, which says:
+- (array)
+    - (enum)
+        - (Member Type)
         - (Element)
 
-- `attributes` (enum)
+##### Examples
 
-    The `attributes` property defines attributes about the given instance of the element, as specified by the `element` property. It MAY be an array of elements, or it MAY be an object. The keys of the object SHOULD be defined element types with valid values for the corresponding element types.
+An Object Type that is an object will look like this example.
 
-    - Members
-        - (object)
-        - array[Element]
+```json
+{
+  "foo": "bar",
+  "hello": "world"
+}
+```
 
-- `content` (enum)
+An Object Type that is an array of Member Types, will look like this. If this structure were to be converting back to a JSON object, it would look like the previous example.
 
-    The `content` property defines the content of the instance of the specified element type. The value MAY be any of the JSON primitive types, an element, or an array of elements.
+```json
+[
+  {
+    "key": "foo",
+    "value": "bar"
+  },
+  {
+    "key": "hello",
+    "value": "world"
+  }
+]
+```
 
-    - Members
-        - (null)
+When a `foo` element is defined that has a string for its content, the element MAY be used instead of a Member Type as an item in the Object Type array.
+
+```json
+[
+  {
+    "element": "foo",
+    "content": "bar"
+  },
+  {
+    "key": "hello",
+    "value": "world"
+  }
+]
+```
+
+#### Member Type (enum)
+
+A Member Type is a structural type that provides a key/value pair to be used with Object Types.
+
+##### Members
+
+- (object)
+    - key (enum, required)
         - (string)
-        - (number)
-        - (boolean)
-        - (array)
-        - (object)
         - (Element)
-        - array[Element]
+    - value (enum, required)
+        - (Null Type)
+        - (String Type)
+        - (Number Type)
+        - (Boolean Type)
+        - (Array Type)
+        - (Object Element)
+- (array)
+    - (Key Element)
+    - (Value Element)
+- (Member Element)
 
-#### Example
+##### Examples
 
-An simple element may look like this, where `foo` is the element type, `id` is an attribute for the `foo` element, and `content` is a string with a value of `bar`.
+When the Member Type is an object, it MAY look like this.
 
 ```json
 {
-  "element": "foo",
-  "attributes": {
-    "id": 4
-  },
-  "content": "bar"
+  "key": "foo",
+  "value": "bar"
 }
 ```
 
-##### XML Representation
+As an object, the key and/or value MAY be Refract elements. In this example, the `value` from the previous example is defined as a String Element. The key MAY also be defined with a Refract Element, though it is shown in this example.
+
+```json
+{
+  "key": "foo",
+  "value": {
+    "element": "string",
+    "content": "bar"
+  }
+}
+```
 
-The example of above would be represented like this in XML.
+When the Member Type is an array, it MUST contain one and only one Key Element along with one and only one Value Element.
 
-```xml
-<foo id="4">bar</foo>
+```json
+[
```

Comment:
So, you have this definition here, which is good per my prior comment, but per prior you also define:

`['foo', {}, {}, 'bar']` which only has meaning in one context as a member, but not anywhere else. So, you would need this example, but that as standalone. When you see this its a member if, means mentally looking at this you are always coupled with asking what is this? An element or member. I know, broken record.